domain/lib/repositories/auth_repository.dart

```diff
@@ -0,0 +1,20 @@
+import 'package:domain/domain.dart';
+
+abstract class AuthRepository {
+  Future<AppUser> signIn({
+    required String email,
+    required String password,
+  });
+
+  Future<AppUser> signInUsingGoogleAcc();
+
+  Future<AppUser> signUp({
```

Comment:
check return types, im not sure that u should return user domain model, because u save user in Hive in this method anyway. Also add SignUp and SignIn models insted of string row in parameters list